src/error.rs

```diff
@@ -0,0 +1,65 @@
+//! The Error types produced by Taffy.
+#[cfg(feature = "std")]
+use core::fmt::{Display, Formatter, Result};
+
+use crate::node::Node;
+
+/// The [`Node`](node::Node) was not found in the [`Taffy`] instance
+#[derive(Debug)]
+pub struct InvalidNode(pub Node);
+
+#[cfg(feature = "std")]
+impl Display for InvalidNode {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        write!(f, "Node {:?} is not in the Taffy instance", self.0)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for InvalidNode {}
+
+/// An error that occurs while trying to access or modify a [`Node`](node::Node)'s children by index.
+#[derive(Debug)]
+pub enum InvalidChild {
+    /// The parent [`Node`](node::Node) does not have a child at `child_index`. It only has `child_count` children
+    ChildIndexOutOfBounds {
+        /// The parent node whose child was being looked up
+        parent: Node,
+        /// The index that was looked up
+        child_index: usize,
+        /// The total number of children the parent has
+        child_count: usize,
+    },
+    /// The [`Node`](node::Node) was not found in the [`Taffy`] instance.
```

Comment:
```suggestion
    /// The parent [`Node`](node::Node) was not found in the [`Taffy`] instance.
```

Reply:
I believe this is correct?

Reply:
yes we can - `Taffy::replace_child_at_index()` if `new_child` isn't valid.